pallene/coder.lua

```diff
@@ -1443,6 +1443,25 @@ gen_cmd["BuiltinMathAbs"] = function(self, cmd, _func)
     return util.render([[ $dst = l_mathop(fabs)($v); ]], { dst = dst, v = v })
 end
 
+gen_cmd["BuiltinMathCeil"] = function(self, cmd, _func)
+    local dst = self:c_var(cmd.dsts[1])
+    local v = self:c_value(cmd.srcs[1])
+    return util.render([[ $dst = (LUA_INTEGER)l_mathop(ceil)($v); ]], { dst = dst, v = v })
```

Comment:
> I believe that casting an float to an integer is undefined behavior if the float is outside of the integer range. We might want to avoid that, because the C compiler can do weird things if it identifies undefined behavior. For example, that could happen if the argument to math.ceil is a suitable floating-point constant...

So for the modf implementation, I created an inline function which ultimately goes through the lua_numbertointeger macro. The side-effect of that is that since I initialized the temporary return value to 0, if lua_numbertointeger cannot convert, the value returned to the user is 0. This at least side-steps undefined behavior in that we can dictate we will return 0 (or whatever integer value we want) for this case. I can make that change to ceil & floor if you wish. 

> One way to avoid this would be to copy what Lua does to the letter (except by possibly raising an error if there is overflow). Alternatively, we could create math functions with diffferent names, without trying to convert to integer.

As I mentioned in my other response, I feel like the Lua reference gives us leeway here, as they do not explicitly dictate what we should return here. I personally don't think it is worth creating different variants of this function right now, just to handle these exceptional cases, especially when the spec doesn't even suggest it.

Raising an error is an interesting idea, but I feel it might be too aggressive since Lua just lets it slide through. I think raising an error here when Lua doesn't may just be obnoxious.